server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/VariableWidthHistogramAggregator.java

```diff
@@ -354,21 +355,22 @@ private void moveLastCluster(int index){
                 clusterSizes.set(index, holdSize);
 
                 // Move the underlying buckets
-                long[] mergeMap = new long[numClusters];
-                for (int i = 0; i < index; i++) {
-                    // The clusters in range {0 ... idx - 1} don't move
-                    mergeMap[i] = i;
-                }
-                for (int i = index; i < numClusters - 1; i++) {
-                    // The clusters in range {index ... numClusters - 1} shift up
-                    mergeMap[i] = i + 1;
-                }
-                // Finally, the new cluster moves to index
-                mergeMap[numClusters - 1] = index;
+                UnaryOperator<Long> mergeMap = new UnaryOperator<Long>() {
+                    @Override
+                    public Long apply(Long i) {
+                       if(i < index) {
+                           // The clusters in range {0 ... idx - 1} don't move
+                           return i;
+                       } else if(i == numClusters - 1) {
+                           // The new cluster moves to index
+                           return (long)index;
+                       } else {
+                           // The clusters in range {index ... numClusters - 1} shift forward
+                           return i + 1;
+                       }
+                    }
+                };
```

Comment:
I'd have tried to write this:
```
mergeBuckets(numClusters, bucket -> {
  if (i < index) {
    // The clusters in range {0 ... idx - 1} don't move
    return 1;
  }
  if (i == numClusters - 1) {
    // The new cluster moves to index
    return i;
  }
  // The clusters in range {index ... numClusters - 1} shift forward
  return i = 1;
});
```

I like the "inline function declaration" form of this because it makes it super obvious that it doesn't escape.

I also like early return instead of `else if`, but that is totally up to you. Its a matter of style and we don't have a standard.

Reply:
I'm not sure if this would work, since this same function is used in the calls to both`BucketsAggregator::mergeBuckets` and `MergingBucketsDeferringCollector::mergeBuckets`.

Reply:
Ah! Well, what you have is just fine too.